src/coreclr/vm/riscv64/cgencpu.h

```diff
@@ -127,6 +127,27 @@ struct FloatArgumentRegisters {
     double  f[8];  // f0-f7
 };
 
+//**********************************************************************
+// Profiling
+//**********************************************************************
+
+#ifdef PROFILING_SUPPORTED
+
+struct PROFILE_PLATFORM_SPECIFIC_DATA
+{
+    void*                  Fp;
+    void*                  Pc;
+    ArgumentRegisters      argumentRegisters;
+    FunctionID             functionId;
+    FloatArgumentRegisters floatArgumentRegisters;
+    void*                  probeSp;
+    void*                  profiledSp;
+    void*                  hiddenArg;
+    UINT64                 flags;
+    // Scratch space to reconstruct struct passed in registers
+    BYTE                   buffer[sizeof(ArgumentRegisters) + sizeof(FloatArgumentRegisters)];
```

Comment:
Is it sufficient for all arguments and return? `CopyStructFromRegisters` can copy argument registers, return registers and stacks. Could you please check again?

Reply:
IMO, it will suffice. For return values we have a separate structure allocated by the `ProfileLeaveNaked` stub. So worst case scenario would be a function with 8 or more arguments of type `struct LongDouble { long; double; }` to use all argument registers for arguments needed to be reconstructed in scratch space.

I'll check the edge-cases where the last struct in registers needs to be partially on the stack, e.g. `func(int i, LongDouble ld1 ... ld8)` but in that case IMO we still fit within the limit.